### ISSUE TITLE
Investigate and officially support hermes engine on iOS

### ISSUE THREAD

**Issue by jennmueng** [getsentry/sentry-react-native#1262]
Support for hermes on iOS was just added in [React Native 0.64-rc2](https://github.com/facebook/react-native/releases/tag/v0.64.0-rc.2) as an opt-in feature, we should investigate this and ensure our SDK fully supports it. We’ve had issues with hermes source maps in the past with Android, now that hermes is being extended to iOS we should ensure that there won’t be any issues.

**Comment by janicduplessis** [comment]
I’m currently using it and it just worked without any changes needed

**Comment by jennmueng** [comment]
So far it doesn't work for me, sending an event gets:
```
 Sentry Logger [Error]: SentryError: Event processing pipeline threw an error, original event will not be sent. Details have been sent as a new event. Reason: TypeError: undefined is not a function

**Comment by jennmueng** [comment]
The error I mentioned in my previous comment that caused Sentry to not send events at all in 0.64 will be fixed by #1274. Next step is to investigate source maps from iOS Hermes

**Comment by jennmueng** [comment]
So far none of the issues that I have run into have been related to Hermes but rather unrelated things such as #1274. So I will close this issue for now.

Please reopen/let us know if you run into any issues with Hermes on iOS.
